### DISCUSSION TITLE
Create an option for `git sl` to allow reverse ordering of commit history

### DISCUSSION THREAD

**Discussion by macintacos** [arxanas/git-branchless#772] [Ideas]
I'm trying to use `git sl` in [`lazygit`](https://github.com/jesseduffield/lazygit) to display commits for a given branch in the main view to the right, but because `git sl` seems to assume you are running this in a terminal session, the most recent commits appear all the way at the bottom of the list, instead of in reverse order (starting with the most recent commit at the top).

Any chance there could be an option introduced on `git sl` that would allow me to generate the smartlog in reverse order?

**Comment by arxanas** [comment]
There is currently no way to do this. I'm happy to accept a PR implementing this.

You might want to check https://github.com/martinvonz/jj, which implements a graph log in both directions, or even the underlying library from Sapling which it uses for the curved style: https://github.com/martinvonz/jj/pull/1051

How are you integrating with `lazygit`?

**Reply by macintacos** [reply]
I just provide `git sl` for the `branchLogCmd` in my `lazygit` configuration here: https://github.com/jesseduffield/lazygit/blob/16802a048e0425182f08cf0e9a2bc31480e9a55d/docs/Config.md?plain=1#L91

Works pretty well, although there's a bug that I was going to file soon - I tend to get fatal errors whenever I switch through branches too quickly 😬 

Thanks for the suggestion, I'll check those other tools out!

**Reply by adzenith** [reply]
There's now a `--reverse` flag that you can pass to git-smartlog
